source/MaterialXCore/Version.cpp

```diff
@@ -1307,7 +1347,7 @@ void Document::upgradeVersion()
                             else
                             {
                                 combineInInput->setConnectedNode(separateNode);
-                                combineInInput->setOutputString(combineInInput->isColorType() ? "outr" : "outx");
+                                combineInInput->setOutputString(inInput->isColorType() ? "outr" : "outx");
```

Comment:
The `combineInInput` is of type `float` and can not be used here to check for color. We still have the original `inInput` that gives us the right answer when queried.